Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/RecordNavPage.xaml.cs

```diff
@@ -40,197 +31,306 @@ namespace Ginger.BusinessFlowsLibNew.AddActionMenu
+            }
+        }
 
-            if (xWinGridUC.comboBoxSelectedValue != null)
+        private void InitMethods()
+        {
+            string targetApp = string.Empty;
+            if (mContext.BusinessFlow.CurrentActivity == null || string.IsNullOrEmpty(mContext.BusinessFlow.CurrentActivity.TargetApplication))
             {
-                if (mWindowExplorerDriver == null)
-                    mWindowExplorerDriver = xWinGridUC.mWindowExplorerDriver;
-                //if (RecordingButton.IsChecked == true)
-                if (isRecording)
+                targetApp = ((ApplicationAgent)mContext.Runner.ApplicationAgents[0]).AppName;
+            }
+            else if (mContext.BusinessFlow.CurrentActivity != null && !string.IsNullOrEmpty(mContext.BusinessFlow.CurrentActivity.TargetApplication))
+            {
+                targetApp = mContext.BusinessFlow.CurrentActivity.TargetApplication;
+            }
+
+            mActivityPlatform = (from x in WorkSpace.Instance.Solution.ApplicationPlatforms
+                                     where x.AppName == targetApp
+                                     select x.Platform).FirstOrDefault();
+            SetControlsVisibility();
+            SetRecordingButtonText();
+            SetMultiplePropertiesGridView();            
+
+            if (xWinGridUC.mContext == null)
+            {
+                xWinGridUC.mContext = mContext; 
+            }
+
+            if (xWinGridUC.mWindowExplorerDriver == null && mWindowExplorerDriver != null)
+            {
+                xWinGridUC.mWindowExplorerDriver = mWindowExplorerDriver;
+            }
+
+            if (xWinGridUC.WindowsComboBox != null)
+            {
+                xWinGridUC.WindowsComboBox.SelectionChanged -= WindowsComboBox_SelectionChanged;
+                xWinGridUC.WindowsComboBox.SelectionChanged += WindowsComboBox_SelectionChanged; 
+            }
+
+            Agent agent = AgentHelper.GetDriverAgent(mContext.BusinessFlow.CurrentActivity, mContext.Runner, mContext);
+            if (agent != null)
+            {
+                agent.PropertyChanged -= Agent_PropertyChanged;
+                agent.PropertyChanged += Agent_PropertyChanged; 
+            }
+        }
+
+        /// <summary>
+        /// Context Property changed event
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void Context_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if(e != null && (e.PropertyName == nameof(BusinessFlow) || e.PropertyName == nameof(Activity)))
+            {
+                if (e.PropertyName == nameof(BusinessFlow))
                 {
-                    StartRecording();
+                    mContext = (Context)sender;
+                    xWinGridUC.mContext = mContext;
                 }
-                else
+
+                InitMethods();
```

Comment:
1- I don't think we should stop recording in case user shifted Activity and the T.app in selected Activity is the same
2- Recording Stop/ Reset is been done only in case of Context BF/Activity change- check what happens if you change the Solution during recording

Reply:
Done